solidity/contracts/bridge/WalletCoordinator.sol

```diff
@@ -0,0 +1,182 @@
+
+    mapping(address => bool) public isProposalSubmitter;
+
+    mapping(bytes20 => uint32) public walletLock;
+
+    event DepositSweepProposalValidityUpdated(
+        uint32 depositSweepProposalValidity
+    );
+
+    event DepositSweepMaxSizeUpdated(uint16 depositSweepMaxSize);
+
+    event ProposalSubmitterAdded(address indexed proposalSubmitter);
+
+    event ProposalSubmitterRemoved(address indexed proposalSubmitter);
+
+    event DepositSweepProposalSubmitted(
+        DepositSweepProposal proposal,
+        address indexed proposalSubmitter
+    );
+
+    modifier onlyProposalSubmitter() {
+        require(
+            isProposalSubmitter[msg.sender],
+            "Caller is not proposal submitter"
+        );
+        _;
+    }
+
+    modifier onlyAfterWalletLock(bytes20 walletPubKeyHash) {
+        require(
+            /* solhint-disable-next-line not-rely-on-time */
+            block.timestamp > walletLock[walletPubKeyHash],
+            "Wallet locked"
+        );
+        _;
+    }
+
+    function initialize(Bridge _bridge) external initializer {
+        __Ownable_init();
+
+        bridge = _bridge;
+        depositSweepProposalValidity = 4 hours;
+        depositSweepMaxSize = 5;
+    }
+
+    function updateDepositSweepProposalValidity(
+        uint32 _depositSweepProposalValidity
+    ) external onlyOwner {
+        depositSweepProposalValidity = _depositSweepProposalValidity;
+        emit DepositSweepProposalValidityUpdated(_depositSweepProposalValidity);
+    }
+
+    function updateDepositSweepMaxSizeUpdated(uint16 _depositSweepMaxSize)
+        external
+        onlyOwner
+    {
+        depositSweepMaxSize = _depositSweepMaxSize;
+        emit DepositSweepMaxSizeUpdated(_depositSweepMaxSize);
+    }
+
+    function addProposalSubmitter(address proposalSubmitter)
+        external
+        onlyOwner
+    {
+        require(
+            !isProposalSubmitter[proposalSubmitter],
+            "This address is already a proposal submitter"
+        );
+        isProposalSubmitter[proposalSubmitter] = true;
+        emit ProposalSubmitterAdded(proposalSubmitter);
+    }
+
+    function removeProposalSubmitter(address proposalSubmitter)
+        external
+        onlyOwner
+    {
+        require(
+            isProposalSubmitter[proposalSubmitter],
+            "This address is not a proposal submitter"
+        );
+        delete isProposalSubmitter[proposalSubmitter];
+        emit ProposalSubmitterRemoved(proposalSubmitter);
+    }
+
+    // TODO: We can also add an ability for the wallet members to unlock on
+    //       their own.
+    function unlockWallet(bytes20 walletPubKeyHash) external onlyOwner {
+        // Just in case, allow the owner to unlock the wallet earlier.
+        walletLock[walletPubKeyHash] = 0;
+    }
+
+    function submitDepositSweepProposal(DepositSweepProposal calldata proposal)
+        external
+        onlyProposalSubmitter
+        onlyAfterWalletLock(proposal.walletPubKeyHash)
```

Comment:
What is the purpose of it?

Reply:
We enforce 1 proposal at a time. Once we add redemptions and moving funds into the mix, that will be super important that only a single operation take place at once.